### PR TITLE
Fix the logic for checking valid dates and tests

### LINKED ISSUE #199: Fix redundant Birthday checks

### PR DESCRIPTION
**Pull request by appleraincoat**
- Fix the issue where invalid birthdays in the future is only checked against whether it is in future year or not, so the same year but future month will still be valid even if it is not
- Fix the tests that test if the date is in the future with relative dates instead of static ones as the tests will fail once that date is reached
- Fix #199

### CODE REVIEW

File: src/main/java/seedu/realodex/model/person/Birthday.java [hunk 1]
```diff
@@ -17,11 +16,12 @@
 public class Birthday {
 
     public static final String INPUT_DATE_PATTERN = "ddMMMyyyy";
-    public static final String MESSAGE_CONSTRAINTS = "Birthday should be in " + INPUT_DATE_PATTERN + " format\nDate "
-            + "should also not be in future years and no earlier than year 1000!";
+    public static final String MESSAGE_CONSTRAINTS = "Birthday should be in " + INPUT_DATE_PATTERN + " format\n"
+            + "A valid date should be given, and it cannot be in the future and no earlier than year 1000.";
```
Comment (jyztintan) [inline src/main/java/seedu/realodex/model/person/Birthday.java:20]: Clear instructions.

File: src/test/java/seedu/realodex/model/person/BirthdayTest.java [hunk 2]
```diff
@@ -31,6 +33,7 @@ public void isValidBirthday() {
 
         // invalid parts
         assertFalse(Birthday.isValidBirthday("29Feb2023")); // not a leap year
+        assertFalse(Birthday.isValidBirthday("31June2023")); // June does not have 31 days
```
Comment (jyztintan) [inline src/test/java/seedu/realodex/model/person/BirthdayTest.java:36]: ok good addition

File: src/test/java/seedu/realodex/model/person/BirthdayTest.java [hunk 3]
```diff
@@ -41,12 +44,23 @@ public void isValidBirthday() {
...
+        // creating future dates
+        LocalDate today = LocalDate.now();
+        LocalDate tomorrow = today.plusDays(1);
+        LocalDate futureDate = today.plusYears(10);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("ddMMyyyy");
+        String tomorrowFormatted = tomorrow.format(formatter);
+        String futureFormatted = futureDate.format(formatter);
+
+        // future dates
+        assertFalse(Birthday.isValidBirthday(tomorrowFormatted)); //cant be in future days
```
Comment (jyztintan) [inline src/test/java/seedu/realodex/model/person/BirthdayTest.java:57]: good thinking!

**Review by jyztintan — APPROVED** [review]
LGTM 👍

File: src/main/java/seedu/realodex/model/person/Birthday.java [hunk 4]
```diff
@@ -67,29 +67,17 @@ public static boolean isValidBirthday(String birthday) {
...
-                // Validate the day (1-31)
-                int day = cal.get(Calendar.DAY_OF_MONTH);
-                if (day >= 1 && day <= 31) {
-                    // Validate the year (e.g., not in the future)
-                    int currentYear = Calendar.getInstance().get(Calendar.YEAR);
-                    int year = cal.get(Calendar.YEAR);
-                    if (year >= 1000 && year <= currentYear) {
-                        return true;
-                    }
-                }
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/model/person/Birthday.java:85]: Thanks for changing, even though its not your function

File: src/test/java/seedu/realodex/model/person/BirthdayTest.java [hunk 5]
```diff
@@ -66,7 +80,7 @@ public void equals() {
     }
 
     @Test
-    public void birthdayDefaultConstructor_equalsDefault() {
+    public void birthdayDefaultConstructor_equalsDefault_test() {
         Birthday defaultBirthday = new Birthday();
```
Comment (UdhayaShan1) [inline src/test/java/seedu/realodex/model/person/BirthdayTest.java:84]: thanks!

**Review by UdhayaShan1 — APPROVED** [review]
LGTM

**Comment by 4llysa** [comment]
Thanks. You can leave my feature to me next time.

**Pull request merged**